stdlib/universe/table_fns.go

```diff
@@ -0,0 +1,445 @@
+			} else {
+				vsSlice = append(vsSlice, values.NewNull(semantic.Float))
+			}
+		case flux.TBool:
+			if vs := cr.Bools(idx); vs.IsValid(i) {
+				vsSlice = append(vsSlice, values.New(vs.Value(i)))
+			} else {
+				vsSlice = append(vsSlice, values.NewNull(semantic.Bool))
+			}
+		case flux.TTime:
+			if vs := cr.Times(idx); vs.IsValid(i) {
+				vsSlice = append(vsSlice, values.New(values.Time(vs.Value(i))))
+			} else {
+				vsSlice = append(vsSlice, values.NewNull(semantic.Time))
+			}
+		default:
+			execute.PanicUnknownType(typ)
+		}
+	}
+	return values.NewArrayWithBacking(flux.SemanticType(typ), vsSlice)
+}
+
+type getRecord struct {
+	*functionValue
+}
+
+func NewGetRecordFunction() values.Value {
+	return &getRecord{}
+}
+
+func (f *getRecord) Type() semantic.Type {
+	return semantic.NewFunctionType(semantic.FunctionSignature{
+		Parameters: map[string]semantic.Type{
+			getRecordTableArg: objects.TableMonoType,
+			getRecordIndexArg: semantic.Int,
+		},
+		Required:     semantic.LabelSet{getRecordTableArg, getRecordIndexArg},
+		PipeArgument: getRecordTableArg,
+		// TODO(affo): this return type should be parameterized by the input types.
+		//  But MonoTypes do not support parametrization.
+		Return: semantic.Object,
+	})
+}
+
+func (f *getRecord) PolyType() semantic.PolyType {
+	return semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			getRecordTableArg: objects.TableType,
+			getRecordIndexArg: semantic.Int,
+		},
+		Required:     semantic.LabelSet{getRecordTableArg, getRecordIndexArg},
+		PipeArgument: getRecordTableArg,
+		// TODO(affo): this return type should be parameterized by the input types. It cannot be TVar,
+		//  because if there is a TVar on the right, there must be one on the left.
+		Return: semantic.Object,
+	})
+}
+
+func (f *getRecord) Function() values.Function {
+	return f
+}
+
+func (f *getRecord) Equal(other values.Value) bool {
+	o, ok := other.(*getRecord)
+	return ok && (f == o)
+}
+
+func (f *getRecord) HasSideEffect() bool {
+	return false
+}
+
+func (f *getRecord) Call(args values.Object) (values.Value, error) {
+	arguments := interpreter.NewArguments(args)
+	var tbl flux.Table
+	if v, err := arguments.GetRequired(getRecordTableArg); err != nil {
+		return nil, err
+	} else if v.Type() != objects.TableMonoType {
+		return nil, fmt.Errorf("unexpected type for %s: want %v, got %v", getRecordTableArg, objects.TableMonoType, v.Type())
+	} else {
+		tbl = v.(*objects.Table).Table
+	}
+
+	rowIdx, err := arguments.GetRequiredInt(getRecordIndexArg)
+	if err != nil {
+		return nil, err
+	}
+
+	var r values.Object
+	if err = tbl.Do(func(cr flux.ColReader) error {
+		if rowIdx < 0 || int(rowIdx) >= cr.Len() {
+			return fmt.Errorf("index out of bounds: %d", rowIdx)
+		}
+		r = objectFromRow(int(rowIdx), cr)
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func objectFromRow(idx int, cr flux.ColReader) values.Object {
```

Comment:
replace this with `execute.GetRow`

Reply:
Thank you, I suppose there was an util for this, but I couldn't find that 😅

Reply:
I can't find `GetRow`. We have `ReadRow` but that doesn't return an `values.Object` actually. I could move this function as an util in `execute` or create a `helpers.go` in package `values` that contains helpers that convert from `ColReader` to `values.Value` instances.

Reply:
hm, yeah. nevermind.